src/name_bazaar/ui/subs.cljs

```diff
@@ -61,14 +63,38 @@
      (path-for route (merge
                       (when my-address
                         {:user/address (str my-address)})
-                      params)))
+                      (if (:user/ens-name params)
+                        (set/rename-keys params {:user/ens-name :user/address})
+                        params))))
     "#" "")))
 
 (reg-sub-raw
  :resolved-route-params
  (fn [db p]
    (reaction
     (let [route-params @(subscribe [:district0x/route-params])]
-      (update
-       route-params
-       :user/address (partial try-resolving-address @db))))))
+      (resolve-params @db route-params)))))
+
+(reg-sub-raw
+ :active-resolved-address
+ (fn [db p]
+   (reaction
+    (let [active-address @(subscribe [:district0x/active-address])
+          resolved-address (try-reverse-resolving-address @db active-address)]
+      (truncated-address resolved-address active-address 20)))))
```

Comment:
Truncation shouldn't be done here. The only place where you use it is for `active-address-select`. That component should do truncation by itself. The best would be if the component does truncation with CSS (class `.ellipsis`), and don't have to mess with truncation in cljs. Please do it that way